docs/development/development_guide.md

```diff
@@ -4,9 +4,37 @@ While TheRock started life as a super-project for doing combined builds and rele
 
 While there is much overlap between using TheRock as a development environment and as a CI/release platform, this document is geared at exposing features and techniques specifically targeted at making ROCm developers more productive. Since development features and approaches are built on an as-needed basis, please consider this a working document that presents approaches that have worked for core developers.
 
+## Overall build architecture
+
+TheRock provides:
+
+- A CMake-based build for ROCm components
+- A setuptools-based build for ROCm Python packages
+- (Coming soon!) Scripts for producing native RPM and DEB packages
+- Scripts for building projects like PyTorch and JAX
+
+![Build architecture](therock_build_architecture.excalidraw.svg)
+
+Note that at each layer of the build, developers can build from source OR fetch prebuilt artifacts/packages from sources like pre-commit CI workflow runs or nightly package releases.
```

Comment:
Style suggestions:
```suggestion
Note that at each layer of the build, developers can build from source _or_ fetch prebuilt artifacts/packages from sources like pre-commit CI workflow runs or nightly package releases.
```

+ `pre-built` vs `prebuilt`:

```suggestion
Note that at each layer of the build, developers can build from source _or_ fetch pre-built artifacts/packages from sources like pre-commit CI workflow runs or nightly package releases.
```
Seems at least in this document we're using `pre-built` but our documentation isn't consistent but we should make a call to harmonize with the diagrams:

* https://github.com/search?q=repo%3AROCm%2FTheRock%20prebuilt&type=code
* https://github.com/search?q=repo%3AROCm%2FTheRock+pre-built&type=code

Reply:
In code, `prebuilt` makes more sense than `pre_built` where `-` characters are not allowed.

I didn't see either in https://www.merriam-webster.com/, but I do see `prebuilt` (no dash) on https://dictionary.cambridge.org/us/dictionary/english/prebuilt.

My web browser spell check doesn't like either.

Let's use `prebuilt`?